pkg/sourcereader/embedded.go

```diff
@@ -103,7 +103,123 @@ func (r EmbeddedSourceReader) GetModuleInfo(modPath string, kind string) (module
 	}
 
 	reader := modulereader.Factory(kind)
-	return reader.GetInfo(modDir)
+	mi, err := reader.GetInfo(modDir)
+	mi.RequiredApis = defaultAPIList(modPath)
+	return mi, err
+}
+
+func defaultAPIList(source string) []string {
+	// API lists at
+	// https://console.cloud.google.com/apis/dashboard and
+	// https://console.cloud.google.com/apis/library
+	staticAPIMap := map[string][]string{
+		"community/modules/compute/SchedMD-slurm-on-gcp-partition": {
+			"compute.googleapis.com",
+		},
+		"community/modules/compute/htcondor-execute-point": {
+			"compute.googleapis.com",
+		},
+		"community/modules/compute/schedmd-slurm-gcp-v5-partition": {
+			"compute.googleapis.com",
+		},
+		"community/modules/database/slurm-cloudsql-federation": {
+			"bigqueryconnection.googleapis.com",
+			"sqladmin.googleapis.com",
+		},
+		"community/modules/file-system/DDN-EXAScaler": {
+			"compute.googleapis.com",
+			"deploymentmanager.googleapis.com",
+			"iam.googleapis.com",
+			"runtimeconfig.googleapis.com",
+		},
+		"community/modules/file-system/Intel-DAOS": {
+			"compute.googleapis.com",
+			"iam.googleapis.com",
+			"secretmanager.googleapis.com",
+		},
+		"community/modules/file-system/nfs-server": {
+			"compute.googleapis.com",
+		},
+		"community/modules/project/new-project": {
+			"admin.googleapis.com",
+			"appengine.googleapis.com",      // optional
```

Comment:
Should we be validating against these two if they are optional?

Reply:
Similar issue to `pubsub` for SchedMD v5 Slurm solutions.

Reply:
I don't want to be in a situation where we are halting execution because of a recommended, but not necessarily required API. The best case scenario in my mind is:
* Error when a required API is not enabled
* Warn when a recommended API is not enable
which of course requires that we track that piece of information alongside the API lists. 

For this case, my preference is to only include hard requirements for now.

Reply:
Removed on `HEAD`.